### PR TITLE
database, usagestats: mark deprecation of direct event_logs recording

### PR DESCRIPTION
**Pull request by bobheadxi**
Part of https://github.com/sourcegraph/sourcegraph/issues/56813 - see https://github.com/sourcegraph/sourcegraph/pull/56868 for docs!

## Test plan

n/a

### CODE REVIEW

File: internal/database/event_logs.go [hunk 1]
```diff
@@ -112,6 +112,7 @@ type EventLogStore interface {
 	// CountUsersWithSetting returns the number of users wtih the given temporary setting set to the given value.
 	CountUsersWithSetting(ctx context.Context, setting string, value any) (int, error)
 
+	// ❗ DEPRECATED: Use event recorders from internal/telemetryrecorder instead.
```
Comment (unknwon) [inline internal/database/event_logs.go:115]:
```suggestion
	// DEPRECATED: Use event recorders from internal/telemetryrecorder instead.
```

Just this, in GoLand, has a visual indicator

CleanShot 2023-09-22 at 13 09 06@2x

Comment (bobheadxi) [reply internal/database/event_logs.go:115]: Unfortunately I'm pretty sure we have a linter that will hard-fail PRs that use "real" deprecated methods, which will probably be really annoying 😅 The emoji makes this more of a "soft" deprecation

Comment (unknwon) [reply internal/database/event_logs.go:115]: The linter only checks for new call sites, no?

We have 100+ docstrings with `// DEPRECATED:`

Comment (bobheadxi) [reply internal/database/event_logs.go:115]: Ah - even then, there may be legitimate uses for using the existing event logs directly still, as we're not quite GA.

I'll opened an issue to follow up with a hard deprecation after GA: https://github.com/sourcegraph/sourcegraph/issues/56973

Comment (unknwon) [reply internal/database/event_logs.go:115]: I guess after all this is fine, more like a FYI 😁

**Review by unknwon — APPROVED** [review]

File: internal/database/event_logs.go (hunk 1 above)
Comment (bobheadxi) [reply internal/database/event_logs.go:115]: vscode has it too 😎

**Pull request merged**